frontend/packages/console-shared/src/components/actions/ActionsHookResolver.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { ExtensionHook, Action } from '@console/dynamic-plugin-sdk';
+
+type ActionsHookResolverProps = {
+  scope: any;
+  useValue: ExtensionHook<Action[]>;
+  onValueResolved: (value: Action[]) => void;
+  onValueError: (error: any) => void;
+};
+
+const ActionsHookResolver: React.FC<ActionsHookResolverProps> = ({
+  useValue,
+  scope,
+  onValueResolved,
+  onValueError,
+}) => {
+  const [value, loaded, loadError] = useValue(scope);
+
+  React.useEffect(() => {
+    if (loaded) onValueResolved(value);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loaded, value]);
+
+  React.useEffect(() => {
+    if (loadError) onValueError(loadError);
+  }, [loadError, onValueError]);
```

Comment:
Don't include `onValueError` in the deps.
You could probably combine both effects, no? Should you be resolving if there is an error?

Reply:
I think keeping it in two separate effects makes sense here. We also don't want to stop the resolution when error occurs because the parent should know what to do in case of an error. The resolver should return a resolved value or `undefined` in any case to complete the loading of all resolvers.

Reply:
Removed `onValueError ` from dependencies.